contracts/compliance/modular/modules/TimeTransfersLimitsModule.sol

```diff
@@ -0,0 +1,220 @@
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+pragma solidity 0.8.17;
+
+import "../IModularCompliance.sol";
+import "../../../token/IToken.sol";
+import "./AbstractModule.sol";
+
+contract TimeTransfersLimitsModule is AbstractModule {
+    /// Struct of transfer Counters
+    struct TransferCounter {
+        uint256 value;
+        uint256 timer;
+    }
+
+    struct Limit {
+        uint32 limitTime;
+        uint256 limitValue;
+    }
+
+    struct IndexLimit {
+        bool attributedLimit;
+        uint8 limitIndex;
+    }
+
+    // Mapping for limit time indexes
+    mapping(address => mapping(uint32 => IndexLimit)) public limitValues;
+
+    /// Mapping for limit time frames
+    mapping(address => Limit[]) public transferLimits;
+
+    /// Mapping for users Counters
+    mapping(address => mapping(address => mapping(uint32 => TransferCounter))) public usersCounters;
+
+    event TimeTransferLimitUpdated(address indexed compliance, uint32 limitTime, uint256 limitValue);
+
+    error LimitsArraySizeExceeded(address compliance, uint arraySize);
+
+    /**
+    *  @dev Set the limit of tokens allowed to be transferred in given time frame.
+    *  @param _limit The limit time and value
+    *  Only the owner of the Compliance smart contract can call this function
+    */
+    function setTimeTransferLimit(Limit calldata _limit) external onlyComplianceCall {
+        if (!limitValues[msg.sender][_limit.limitTime].attributedLimit && transferLimits[msg.sender].length >= 4) {
+            revert LimitsArraySizeExceeded(msg.sender, transferLimits[msg.sender].length);
+        }
+
+        if (!limitValues[msg.sender][_limit.limitTime].attributedLimit && transferLimits[msg.sender].length < 4) {
+            transferLimits[msg.sender].push(_limit);
+            limitValues[msg.sender][_limit.limitTime] = IndexLimit(true, uint8(transferLimits[msg.sender].length) - 1);
+        } else {
+            transferLimits[msg.sender][limitValues[msg.sender][_limit.limitTime].limitIndex] = _limit;
+        }
+
+        emit TimeTransferLimitUpdated(msg.sender, _limit.limitTime, _limit.limitValue);
+    }
+
+    /**
+     *  @dev See {IModule-moduleTransferAction}.
+     */
+    function moduleTransferAction(address _from, address /*_to*/, uint256 _value) external override onlyComplianceCall {
```

Comment:
why is _to commented?

Reply:
We don't use that parameter in this module, and the linter gives an error when there is an unused parameter. We can also remove the parameter name comment (only the parameter type remains in this case, but it may reduce the readability of the code)